inference-engine/src/vpu/graph_transformer/src/stages/strided_slice.cpp

```diff
@@ -11,22 +11,30 @@ namespace vpu {
 
 namespace {
 
-int maskStrToInt(std::string mask) {
-    int idx = 0, result = 0;
+int maskStrToInt(std::string mask, int elemCount) {
+    std::vector<int> elemsToPack;
 
     for (const auto& character : mask) {
-        if (character == ',') continue;
-
-        if (idx++ > 0) {
-            result <<= 1;
-        }
-        if (character == '1') {
-            result = result | 1;
-        } else if (character != '0') {
-            VPU_THROW_FORMAT("Unsupported mask value: only 0 or 1 are supported, but got {} instead", character);
+        switch (character) {
+            case ',':
+                continue;
+            case '1':
+                elemsToPack.push_back(1);
+                break;
+            case '0':
+                elemsToPack.push_back(0);
+                break;
+            default:
+                VPU_THROW_FORMAT("Unsupported mask value: only 0 or 1 are supported, but got {} instead", character);
         }
     }
 
+    int result = 0;
+    int idx = 0;
+    for (auto it = elemsToPack.rbegin(); it != elemsToPack.rend(); it++) {
+        result = idx++ > 0 ? (result << 1) | *it : result | *it;
+    }
+
```

Comment:
It could be related to my question above - why do you need to run in reverse order here?

Note: `result` must be unsigned type

```cpp
std::uint32_t result = 0;
for (std::size_t i = 0; i < elemsToPack.size(); ++i) {
    result |= elemstoPack[i];
    if (i < elemsToPack.size() - 1) {
        result <<= 1;
    }
}
```

Reply:
Made result `uint32_t` + refactored.